api/graphql/tests/test_reservation_units.py

```diff
@@ -104,3 +110,134 @@ def test_should_error_when_not_found_by_pk(self):
         assert_that(errors[0].get("message")).is_equal_to(
             "No ReservationUnit matches the given query."
         )
+
+    @override_settings(HAUKI_ORIGIN_ID="1234", HAUKI_API_URL="url")
+    @mock.patch("opening_hours.utils.opening_hours_client.get_opening_hours")
+    @mock.patch("opening_hours.hours.make_hauki_request")
+    def test_opening_hours(self, mock_periods, mock_opening_times):
+        mock_opening_times.return_value = get_mocked_opening_hours(
+            self.reservation_unit.uuid
+        )
+        mock_periods.return_value = get_mocked_periods()
+        query = (
+            f"{{\n"
+            f"reservationUnitByPk(pk: {self.reservation_unit.id}) {{\n"
+            f"id\n"
+            f'openingHours(periods:true openingTimes:true startDate:"2020-01-01" endDate:"2022-01-01")'
+            f"{{openingTimes{{date}} openingTimePeriods{{timeSpans{{startTime}}}}"
+            f"}}"
+            f"}}"
+            f"}}"
+        )
+        response = self.query(query)
+        content = json.loads(response.content)
+        assert_that(content.get("errors")).is_none()
+        assert_that(
+            content.get("data")
+            .get("reservationUnitByPk")
+            .get("openingHours")
+            .get("openingTimePeriods")[0]
+            .get("timeSpans")
+        ).is_not_empty()
+
+        assert_that(
+            content.get("data")
+            .get("reservationUnitByPk")
+            .get("openingHours")
+            .get("openingTimes")
+        ).is_not_empty()
+
+
+def get_mocked_opening_hours(uuid):
+    resource_id = f"{settings.HAUKI_ORIGIN_ID}:{uuid}"
+    return [
+        {
+            "resource_id": resource_id,
+            "date": datetime.date(2020, 1, 1),
+            "times": [
+                TimeElement(
+                    start_time=datetime.time(hour=10),
+                    end_time=datetime.time(hour=22),
+                    end_time_on_next_day=False,
+                    resource_state=State.WITH_RESERVATION,
+                    periods=[1, 2, 3, 4],
+                ),
+            ],
+        },
+        {
+            "resource_id": resource_id,
+            "date": datetime.date(2020, 1, 2),
+            "times": [
+                TimeElement(
+                    start_time=datetime.time(hour=10),
+                    end_time=datetime.time(hour=22),
+                    end_time_on_next_day=False,
+                    resource_state=State.WITH_RESERVATION,
+                    periods=[
+                        1,
+                    ],
+                ),
+            ],
+        },
+    ]
+
+
+def get_mocked_periods():
```

Comment:
Quite a lot of test data setup but i guess it's pretty unavoidable.

Reply:
yup its kinda bloating but, we expect this kind of data to be fetched